sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/DownloadResponseMockFlux.java

```diff
@@ -31,15 +31,16 @@
     private HTTPGetterInfo info;
     private ByteBuffer scenarioData;
 
-    DownloadResponseMockFlux(int scenario) {
+    DownloadResponseMockFlux(int scenario, APISpec apiSpec) {
         this.scenario = scenario;
+
         switch (this.scenario) {
             case DR_TEST_SCENARIO_SUCCESSFUL_ONE_CHUNK:
-                this.scenarioData = APISpec.getRandomData(512 * 1024);
+                this.scenarioData = apiSpec.getRandomData(512 * 1024);
                 break;
             case DR_TEST_SCENARIO_SUCCESSFUL_MULTI_CHUNK:
             case DR_TEST_SCENARIO_SUCCESSFUL_STREAM_FAILURES:
-                this.scenarioData = APISpec.getRandomData(1024);
+                this.scenarioData = apiSpec.getRandomData(1024);
```

Comment:
I think the static method is enough. It is weird to take in the APISpec

Reply:
The method can't be static as it requires test context to properly record the data.

Reply:
Not a big blocker for this. Might be better to add the randomData method in TestRandomNamer